### DISCUSSION TITLE
Offline Repository access in Mobile Github app

### DISCUSSION THREAD

**Discussion by Koushikon** [community/community#7365] [Mobile] [answered]
In mobile sometimes we need to access the docs and repositories not to change but to read.

With this in mind mobile apps repos offline access features would be great.

Many repository has thousands of files and that would lead to performance issue So we it would be better to use `.obb` files games has this way accessing would be faster.

**Comment by brianlovin** [comment, marked as answer]
Cool idea, thanks for the note @Koushikon — I've opened a tracking issue for our team to talk about offline access. Right now it's not prioritized, but hopefully we can visit this idea soon!

**Reply by Koushikon** [reply]
Thanks and it would be great to Have offline access inside github app.

**Reply by tomsdev** [reply]
@brianlovin how can I subscribe to that issue? I'd like to follow when offline will be implemented thanks

**Reply by brianlovin** [reply]
It's an internal issue unfortunately, but it's still not on our roadmap if that's a helpful status update 😊

**Reply by shammsauce** [reply]
Any up date on this possibly making it to the roadmap? Would be an awesome feature.

**Reply by YevheniySolovey** [reply]
Hi @brianlovin!
Do you have any update on that topic? It has been a while from the last reply :)

**Reply by okmatija** [reply]
I would love to have this feature! I want to read code when I'm on the subway with no internet 😍

**Reply by ArchBase** [reply]
I need this for exam tomorrow at a weak internet zone

**Reply by caleb-allen** [reply]
I'm with @okmatija, I'd really love to review PRs or my own code after a day of work while I'm on the London Underground but have to resort to git itself. Would love to have this directly in GitHub! Even if it were barebones and simple.

**Reply by mi7ak2020-stack** [reply]
> Cool idea, thanks for the note @Koushikon — I've opened a tracking issue for our team to talk about offline access. Right now it's not prioritized, but hopefully we can visit this idea soon!

**Comment by gcolbourn** [comment]
This would be great for the purposes of archiving important information for use in the event of a global catastrophe (i.e. if thousands or millions have a repository of such information downloaded to their phone for offline viewing. And with github the repository can be kept up to date automatically, right up to when the internet goes down!)

**Reply by Koushikon** [reply]
Yes, It's good to have this feature

**Reply by albrtkinn** [reply]
> This would be great for the purposes of archiving important information for use in the event of a global catastrophe (i.e. if thousands or millions have a repository of such information downloaded to their phone for offline viewing. And with github the repository can be kept up to date automatically, right up to when the internet goes down!)

**Reply by albrtkinn** [reply]
> Yes, It's good to have this feature

**Reply by Koushikon** [reply]
Yes, Its like how Torrent system works here for sharing Public repos.

**Comment by albrtkinn** [comment]
> Cool idea, thanks for the note @Koushikon — I've opened a tracking issue for our team to talk about offline access. Right now it's not prioritized, but hopefully we can visit this idea soon!

**Comment by seyma-si** [comment]
Telegram

**Comment by akashe** [comment]
Any updates on this feature? Sometimes, when I am travelling I don't have access to internet but I am thinking about a particular problem. So atleast being able to view code offline during that time is a great plus. The feature need not be available for all repositories but atleast for a few selected one whose copy can be kept locally also.

**Reply by Koushikon** [reply]
The use cases of this feature are many. It's exciting, right?

**Comment by ad268** [comment]
🇲🇽

**Comment by Ductini2703** [comment]
[]()

**Comment by gith-cmd** [comment]
> In mobile sometimes we need to access the docs and repositories not to change but to read.
> 
> With this in mind mobile apps repos offline access features would be great.
> 
> Many repository has thousands of files and that would lead to performance issue So we it would be better to use `.obb` files games has this way accessing would be faster.
> 
>

**Comment by varolomer** [comment]
Any updates on this feature? It would be really great to have the possibility to examine code or study docs while traveling etc.

**Comment by victor-marino** [comment]
I second this, and can provide another use case.

I'm using a GitHub repository as my usual note taking system (plain markdown files). I'd love to be able to just open the app on my phone, even when I'm offline, write down whatever I want to keep in my notes, and have it automatically sync to the repo whenever my phone is online again.

**Reply by ryu577** [reply]
We have an Obsidian user here.

**Reply by victor-marino** [reply]
Nope, I actually ended up moving away from Obsidian a long time ago, as I got a bit tired of depending on their apps and integrations. Instead I just use VSCode with a markdown extension on my PC, and the GitHub app on my phone. But it'd be nice to get that offline functionality as well for quick note taking when not online.

**Reply by 2103baxterb-boop** [reply]
9~~~~

**Comment by tobybell** [comment]
I’d also love to see this feature! Would love to be able to refer to saved repos while traveling without internet access.

**Comment by SpeedyDoggo** [comment]
I had assumed half the point of the app would be to have the files offline, but I guess not. Uninstalling again :( Hope this actually gets added soon

**Comment by juanpyusun** [comment]
just passing by, there are people interested in this feature yet, don't forget us <3

**Comment by hassan-jalil** [comment]
..

**Comment by vinayakvb** [comment]
Its 2026, people are still waiting for this feature. Please help us!

**Comment by Scooterking38** [comment]
Pls guys, listen to the community and just add this feature already.

**Reply by opentyler** [reply]
I think it's safe to assume they're not going to implement this. Probably better to close this so people have clarity and can move to other apps that can provide the functionality we're users are looking for.

**Reply by Koushikon** [reply]
I am starting to think its true. I don't know where to start if i want to build this. If anyone has the idea please share I am up for it.

**Reply by Scooterking38** [reply]
think what is true?

**Reply by Koushikon** [reply]
They are not going to implement this feature.

**Reply by Scooterking38** [reply]
you cant build it because github is closed source

**Comment by Annallon** [comment]
Offline mode seems like it would be extremely beneficial for those with unstable or no consistent access to internet. Why has this still not been implemented?

**Comment by rd22-cell** [comment]
Hello

**Comment by Munashe-Njanji** [comment]
This feature will be a game changer no matter what!
